static/app/utils.tsx

```diff
@@ -297,14 +301,15 @@ export function isWebpackChunkLoadingError(error: Error): boolean {
   );
 }
 
-export function deepFreeze<T>(object: T) {
+export function deepFreeze<T extends Record<any, any>>(object: T): Readonly<T> {
   // Retrieve the property names defined on object
   const propNames = Object.getOwnPropertyNames(object);
   // Freeze properties before freezing self
-  for (const name of propNames) {
-    const value = object[name];
+  for (let i = 0; i < propNames.length; i++) {
```

Comment:
I changed this because for of is usually ~50% slower than regular loop and since we are iterating over unknown inputs, may as well get the most of it

Reply:
Did you get that 50% from a microbenchmark or is it an actual 50% (a lot of the recommendations about performant have been rendered moot over the last few years)? I generally avoid playing with syntax to make the js engine change behaviour, v8 and others can change perf of specific implementations under the hood. I'd say if our `n` is high enough to be a concern then we have a problem elsewhere.

Reply:
@k-fish yeah, this was something I tested fairly recently with the flamegraph draw code where we used to use for-of and switched to (I cant find the benchmark, but iirc n was ~5k). That said I agree with you, we dont really care about this imo, most problems are in other places and I probably just unnecessarily touched this (I'll revert back). Iirc the main perf benefit is that a regular for loop does not use the iterator object that for of does, but that's internals we shouldnt care about imo.